src/realm/object-store/results.cpp

```diff
@@ -970,6 +970,16 @@ Results Results::distinct(std::vector<std::string> const& keypaths) const
     return distinct({std::move(column_keys)});
 }
 
+Results Results::filter_by_method(std::function<bool(const Obj&)>&& predicate) const
+{
+    DescriptorOrdering new_order = m_descriptor_ordering;
+    new_order.append_filter(FilterDescriptor(std::move(predicate)));
```

Comment:
I'm not quite sure about the approach with adding it into descriptors. There is already "Results filter(Query&& q)" which acts as and_query. Wouldn't it be better to make it part of Query? Is not appropriate? What's your thinking originally about adding it into DescriptorOrdering?

Reply:
I would rather go in the opposite direction to also be able to add a QueryDescriptor. I see the descriptors as different stages in a query process.

Reply:
I see the logic behind it, although it becomes quite confusing then. Filtering methods defined on top on TableView and Results, and not on the Query class itself. So the stages in 'query process' are obvious from api standpoint. Maybe some api redesign is needed to outline this shift. Would've been great to have some ticket describing the problem and potential changes for this.